drivers/drivers.go

```diff
@@ -0,0 +1,98 @@
+// Package drivers abstracts different object storages, such as local, s3, ipfs
+package drivers
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/net"
+)
+
+// Storages configured at start (used by broadcaster)
+var Storages []OSDriver
+
+// ExternalStorage own external storage, if configured
+var ExternalStorage OSDriver
+
+// LocalStorage single instance of local storage
+var LocalStorage *localOS
+
+// OSDriver common interface for Object Storage
+type OSDriver interface {
+	SaveData(jobID int64, streamID, name string, data []byte) (*net.TypedURI, error)
```

Comment:
@j0sh I've left some comments in #588.
`JobID` and `streamID` is here, because I wanted to hide logic that make directory name for data inside driver, but, probably, we can changes this to `SaveData(dirName, name string...)`